Neos.Flow/Classes/Mvc/Routing/Dto/ResolveContext.php

```diff
@@ -62,6 +62,15 @@ public function __construct(UriInterface $baseUri, array $routeValues, bool $for
         $this->routeValues = $routeValues;
         $this->forceAbsoluteUri = $forceAbsoluteUri;
         $this->uriPathPrefix = $uriPathPrefix;
+
+        // Only add base uri path for absolute uri, in case of relative uri the uri has to be relative to the given base uri
+        if ($forceAbsoluteUri) {
+            $this->uriPathPrefix = '/' . ltrim($this->uriPathPrefix, '/');
+
+            if ($baseUri->getPath() !== '') {
+                $this->uriPathPrefix = rtrim($baseUri->getPath(), '/') . $this->uriPathPrefix;
```

Comment:
Ok, so I don't understand this really and think we should revert this whole change and rethink what happens. This was a simple value object before, now it does some internal magic, that doesn't line up with what happens in `\Neos\Flow\Mvc\Routing\Router::resolve` because there the uriPathPrefix is applied to the baseUri anyway, which will result in duplication in `\Neos\Flow\Mvc\Routing\Dto\UriConstraints::applyTo`. The tests do not expose this (yet) because the failing scenario is an installation in a subdirectory of the webhost, because that will have a baseUri with the subdirectory + a script request path (which then becomes a uriPathPrefix) which is the subdirectory. So that subdirectory gets preprended to the path twice. We should make sure to solve this here in another place.